File: server/apollo/src/query-statement.ts
import { loadBudget } from '@tally/lib/data/loader';
import {
  GqlStatement,
  QueryStatementArgs
} from './types';
import { toGqlStatement } from './to-gql';

export async function buildStatement(_: any, args: QueryStatementArgs): Promise<GqlStatement> {
  const startTimeMs: number = Date.now();
  const payload = await loadBudget();
  const statement = payload.statements.get(args.account)?.get(args.month.toString());
  if (!statement) {
    throw new Error(
      `Did not find statement for ${args.owner} ${args.account} ${args.month}`
    );
  }
  const result = toGqlStatement(statement);
  console.log(`gql statement in ${Date.now() - startTimeMs}ms`);
  return result;
}

File: server/kt/src/main/kotlin/schema/QueryStatement.kt
package com.cherba29.tally.schema

import com.cherba29.tally.data.DataPayload
import io.github.oshai.kotlinlogging.KotlinLogging
import kotlin.time.Clock

fun buildStatement(payload: DataPayload, owner:String, account: String, month: String): GqlStatement {
  val startTimeMs: Long = Clock.System.now().toEpochMilliseconds()

  val statement = payload.statements[account]?.get(month)
  if (statement == null) {
    throw NotFoundException("Did not find statement for $owner $account $month")
  }
  val result = statement.toGql()
  logger.info { "gql statement in ${Clock.System.now().toEpochMilliseconds() - startTimeMs}ms" }
  return result
}

private val logger = KotlinLogging.logger {}
